polygeist/tools/cgeist/driver.cc

```diff
@@ -287,8 +287,6 @@ static int canonicalize(mlir::MLIRContext &Ctx,
   OptPM.addPass(polygeist::createLoopRestructurePass());
   OptPM.addPass(polygeist::replaceAffineCFGPass());
   OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
-  if (ScalarReplacement)
-    PM.addNestedPass<func::FuncOp>(mlir::createAffineScalarReplacementPass());
```

Comment:
This is not needed and causes confusion, as we were simply adding two `AffineScalarReplacementPass` one right after another one. Notice that it was adding to `PM`, so it is added after all passes added in `OptPM`.